libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1588,6 +1589,7 @@ bool ExpressionCompiler::visit(IndexAccess const& _indexAccess)
 			{
 				ArrayUtils(m_context).accessIndex(arrayType, true);
 				if (arrayType.baseType()->isValueType())
+					// TODO if ABIEncoderV2 is enabled, run validate?
```

Comment:
Yes, it should - both this and the one above are probably the only places that need to be changed for arrays and structs.

Reply:
Implemented.